x-pack/solutions/security/plugins/security_solution/server/endpoint/routes/suggestions/index.ts

```diff
@@ -58,27 +59,54 @@ export function registerEndpointSuggestionsRoutes(
       withEndpointAuthz(
         { any: ['canWriteEventFilters'] },
         endpointContext.logFactory.get('endpointSuggestions'),
-        getEndpointSuggestionsRequestHandler(config$, getLogger(endpointContext))
+        getEndpointSuggestionsRequestHandler(config$, endpointContext)
       )
     );
 }
 
 export const getEndpointSuggestionsRequestHandler = (
   config$: Observable<ConfigSchema>,
-  logger: Logger
+  endpointContext: EndpointAppContext
 ): RequestHandler<
   TypeOf<typeof EndpointSuggestionsSchema.params>,
   never,
   EndpointSuggestionsBody,
   SecuritySolutionRequestHandlerContext
 > => {
   return async (context, request, response) => {
+    const logger = getLogger(endpointContext);
```

Comment:
General comment: our route handler should all be wrapped in a `try{...} catch()` block and use our [Route error handler](https://github.com/elastic/kibana/blob/70d6f40007b7c4a16335ff911270ff2392bbf7e0/x-pack/solutions/security/plugins/security_solution/server/endpoint/routes/error_handler.ts#L20) in the catch block.

Reply:
Good spot, thanks.  Wrapped - 8cd26fe02787c0eefcb7104bf803d3ad044e7206